### PR TITLE
chore(Various): updated CSS gap vars to use semantic tokens (part 1)

### PR DESCRIPTION
**Pull request by thatblindgeye**
Towards #6715

Decided to put this PR up that only touches some files rather than a huge PR, partially because there's some questions to maybe iron out. I added a few tokens to the local file that I'm wondering if we should add to Figma, and I left some comments about replacement tokens being a different gap than the current gap or CSS vars that didn't seem to have an applicable gap token (at least based on name).

### CODE REVIEW

**Review by srambach — COMMENTED** [review]
A few questions based on inspection of what the variable/token controls. I'll leave the decisions about whether spacing increases/decreases to design :-) 

I'd love to move the local tokens out to figma, though each is only used once. Will there be more? Also, are there others to create where "no applicable spacer" exists?

File: src/patternfly/components/Content/content.scss [hunk 1]
```diff
@@ -80,12 +80,12 @@
...
   // Lists
-  --#{$content}--list--Gap: var(--pf-t--global--spacer--sm);
+  --#{$content}--list--Gap: var(--pf-t--global--spacer--gap--group--vertical);
   --#{$content}--list--PaddingInlineStart: var(--pf-t--global--spacer--lg);
   --#{$content}--list--nested--MarginBlockStart: var(--#{$content}--list--Gap);
   --#{$content}--ul--ListStyle: var(--pf-t--global--list-style);
-  --#{$content}--dl--ColumnGap: var(--pf-t--global--spacer--2xl);
-  --#{$content}--dl--RowGap: var(--pf-t--global--spacer--md);
+  --#{$content}--dl--ColumnGap: var(--pf-t--global--spacer--gap--group-to-group--horizontal);
+  --#{$content}--dl--RowGap: var(--pf-t--global--spacer--gap--group-to-group--vertical); // Increases the current gap
```
Comment (mcoker) [inline src/patternfly/components/Content/content.scss:88]: Met with @lboehling today and we looked at the content `<dl>` and description-list component (for another PR - https://github.com/patternfly/patternfly/pull/6953/files) and the row-gap here is good but the column gap should be  `--pf-t--global--spacer--gap--group--horizontal`

```suggestion
  --#{$content}--dl--ColumnGap: var(--pf-t--global--spacer--gap--group--horizontal);
  --#{$content}--dl--RowGap: var(--pf-t--global--spacer--gap--group-to-group--vertical);
```

File: src/patternfly/components/DescriptionList/description-list.scss [hunk 2]
```diff
@@ -3,22 +3,22 @@
 $pf-v6-c-description-list--breakpoint-map: build-breakpoint-map("base", "sm", "md", "lg", "xl", "2xl");
 
 :where(:root, .#{$description-list}) {
-  --#{$description-list}--RowGap: var(--pf-t--global--spacer--lg);
-  --#{$description-list}--ColumnGap: var(--pf-t--global--spacer--lg);
+  --#{$description-list}--RowGap: var(--pf-t--global--spacer--gap--group-to-group--vertical);
+  --#{$description-list}--ColumnGap: var(--pf-t--global--spacer--gap--group-to-group--horizontal); // Increases current gap
```
Comment (mcoker) [inline src/patternfly/components/DescriptionList/description-list.scss:7]: I think these are both good 👍

File: src/patternfly/components/DescriptionList/description-list.scss [hunk 3]
```diff
@@ -3,22 +3,22 @@
...
   --#{$description-list}--GridTemplateColumns--count: 1;
   --#{$description-list}--GridTemplateColumns--width: 1fr;
   --#{$description-list}--GridTemplateColumns--min: 0;
 
   // group
-  --#{$description-list}__group--RowGap: var(--pf-t--global--spacer--sm);
-  --#{$description-list}__group--ColumnGap: var(--pf-t--global--spacer--sm);
+  --#{$description-list}__group--RowGap: var(--pf-t--global--spacer--gap--group--vertical);
+  --#{$description-list}__group--ColumnGap: var(--pf-t--global--spacer--gap--group--horizontal); // Increases current Gap
```
Comment (mcoker) [inline src/patternfly/components/DescriptionList/description-list.scss:14]: And these are good 👍

**Review by mcoker — COMMENTED** [review]
Just a little bit of feedback on `<dl>` spacing from another review today of https://github.com/patternfly/patternfly/pull/6953/. There is a small update we can make here to the content component `<dl>` to match #6953. That change will also make the content component `<dl>` match the spacing in the horizontal description list component.

File: src/patternfly/components/DualListSelector/dual-list-selector.scss [hunk 4]
```diff
@@ -66,7 +66,7 @@ $pf-v6-c-dual-list-selector__item--MaxNesting: 10;
   // Controls
   --#{$dual-list-selector}__controls--PaddingInlineEnd: var(--pf-t--global--spacer--md);
   --#{$dual-list-selector}__controls--PaddingInlineStart: var(--pf-t--global--spacer--md);
-  --#{$dual-list-selector}__controls--Gap: var(--pf-t--global--spacer--sm);
+  --#{$dual-list-selector}__controls--Gap: var(--pf-t--global--spacer--gap--group--vertical);
```
[the remark this answers is not in the document]
Comment (thatblindgeye) [reply src/patternfly/components/DualListSelector/dual-list-selector.scss:69]: Update to --pf-t--global--spacer--gap--action-to-action--plain

File: src/patternfly/components/Form/form.scss [hunk 5]
```diff
@@ -85,15 +85,15 @@ $pf-v6-c-form--m-horizontal--breakpoint-map: build-breakpoint-map("sm", "md", "l
...
   --#{$form}__group-label-info--FontSize: var(--pf-t--global--font--size--body--sm);
-  --#{$form}--m-horizontal__group-label--m-info--Gap: var(--pf-t--global--spacer--sm);
+  --#{$form}--m-horizontal__group-label--m-info--Gap: var(--pf-t--global--spacer--gap--text-to-element--default);
 
   // Group control
   --#{$form}__group-control--m-inline--child--MarginInlineEnd: var(--pf-t--global--spacer--lg);
   --#{$form}__group-control__helper-text--MarginBlockEnd: var(--pf-t--global--spacer--xs);
-  --#{$form}__group-control--Gap: var(--pf-t--global--spacer--sm);
-  --#{$form}__group-control--m-stack--Gap: var(--pf-t--global--spacer--sm);
+  --#{$form}__group-control--Gap: var(--pf-t--global--spacer--gap--group--vertical);
```
[the remark this answers is not in the document]
Comment (thatblindgeye) [reply src/patternfly/components/Form/form.scss:95]: Removed

File: src/patternfly/components/Form/form.scss [hunk 6]
```diff
@@ -112,7 +112,7 @@ $pf-v6-c-form--m-horizontal--breakpoint-map: build-breakpoint-map("sm", "md", "l
 
   // Section
   --#{$form}__section--MarginBlockStart: var(--pf-t--global--spacer--xl);
-  --#{$form}__section--Gap: var(--pf-t--global--spacer--md);
+  --#{$form}__section--Gap: var(--pf-t--global--spacer--gap--group--horizontal);
```
[the remark this answers is not in the document]
Comment (thatblindgeye) [reply src/patternfly/components/Form/form.scss:115]: Updated to use the form--gridgap var per discussion in meeting

**Comment by thatblindgeye** [comment]
[BackstopJS Report Gap Tokens.pdf](https://github.com/user-attachments/files/16854212/BackstopJS.Report.Gap.Tokens.pdf)

cc @mcoker

File: src/patternfly/components/ClipboardCopy/clipboard-copy.scss [hunk 7]
```diff
@@ -21,13 +21,13 @@
...
-  --#{$clipboard-copy}__group--Gap: var(--pf-t--global--spacer--xs);
+  --#{$clipboard-copy}__group--Gap: var(--pf-t--global--spacer--gap--control-to-control--default);
 
   // Inline
   --#{$clipboard-copy}--m-inline--PaddingInlineStart: var(--pf-t--global--spacer--xs);
   --#{$clipboard-copy}--m-inline--PaddingInlineEnd: var(--pf-t--global--spacer--xs);
   --#{$clipboard-copy}--m-inline--BackgroundColor: var(--pf-t--global--background--color--secondary--default);
-  --#{$clipboard-copy}__actions--Gap: var(--pf-t--global--spacer--sm);
+  --#{$clipboard-copy}__actions--Gap: var(--pf-t--global--spacer--gap--action-to-action--default);
   --#{$clipboard-copy}__actions--MarginInlineStart: var(--pf-t--global--spacer--xs);
```
Comment (mcoker) [inline src/patternfly/components/ClipboardCopy/clipboard-copy.scss:31]: Confirmed with @lboehling we can use the new var you created `--pf-t--global--spacer--gap--text-to-element--compact` here

```suggestion
  --#{$clipboard-copy}__actions--MarginInlineStart: var(--pf-t--global--spacer--gap--text-to-element--compact);
```

File: src/patternfly/components/ClipboardCopy/clipboard-copy.scss [hunk 8]
```diff
@@ -21,13 +21,13 @@
...
   // Group
-  --#{$clipboard-copy}__group--Gap: var(--pf-t--global--spacer--xs);
+  --#{$clipboard-copy}__group--Gap: var(--pf-t--global--spacer--gap--control-to-control--default);
 
   // Inline
   --#{$clipboard-copy}--m-inline--PaddingInlineStart: var(--pf-t--global--spacer--xs);
   --#{$clipboard-copy}--m-inline--PaddingInlineEnd: var(--pf-t--global--spacer--xs);
   --#{$clipboard-copy}--m-inline--BackgroundColor: var(--pf-t--global--background--color--secondary--default);
-  --#{$clipboard-copy}__actions--Gap: var(--pf-t--global--spacer--sm);
+  --#{$clipboard-copy}__actions--Gap: var(--pf-t--global--spacer--gap--action-to-action--default);
```
Comment (mcoker) [inline src/patternfly/components/ClipboardCopy/clipboard-copy.scss:30]:
```suggestion
  --#{$clipboard-copy}__actions--Gap: var(--pf-t--global--spacer--gap--action-to-action--plain);
```

File: src/patternfly/components/CodeEditor/code-editor.scss [hunk 9]
```diff
@@ -3,7 +3,7 @@
 @include pf-root($code-editor) {
   // controls
   --#{$code-editor}__controls--PaddingInlineStart: var(--pf-t--global--spacer--sm);
-  --#{$code-editor}__controls--Gap: var(--pf-t--global--spacer--xs);
+  --#{$code-editor}__controls--Gap: var(--pf-t--global--spacer--gap--control-to-control--default);
```
Comment (mcoker) [inline src/patternfly/components/CodeEditor/code-editor.scss:6]:
```suggestion
  --#{$code-editor}__controls--Gap: var(--pf-t--global--spacer--gap--action-to-action--plain);
```

File: src/patternfly/components/Content/content.scss [hunk 10]
```diff
@@ -82,7 +82,7 @@
   --#{$content}--blockquote--BorderInlineStartWidth: var(--pf-t--global--border--width--300); // TODO semantic value?
 
   // Lists
-  --#{$content}--list--Gap: var(--pf-t--global--spacer--sm);
+  --#{$content}--list--Gap: var(--pf-t--global--spacer--gap--group--vertical);
```
Comment (mcoker) [inline src/patternfly/components/Content/content.scss:85]: Chatting with @lboehling we may look to add some content specific tokens in the future, so to be consistent with other content elements (description list excluded) let's update this to be the generic t-shirt spacer for now.

```suggestion
  --#{$content}--list--Gap: var(--pf-t--global--spacer--sm);
```

File: src/patternfly/components/Form/form.scss [hunk 11]
```diff
@@ -26,10 +26,10 @@ $pf-v6-c-form--m-horizontal--breakpoint-map: build-breakpoint-map("sm", "md", "l
...
 @include pf-root($form) {
-  --#{$form}--GridGap: var(--pf-t--global--spacer--lg);
+  --#{$form}--GridGap: var(--pf-t--global--spacer--gap--group-to-group--vertical);
 
   // Group
-  --#{$form}__group--Gap: var(--pf-t--global--spacer--sm);
+  --#{$form}__group--Gap: var(--pf-t--global--spacer--gap--group--vertical);
   --#{$form}__group--m-action--MarginBlockStart: var(--pf-t--global--spacer--xl);
   --#{$form}--m-horizontal__group-label--md--GridColumnWidth: #{pf-size-prem(150px)};
   --#{$form}--m-horizontal__group-label--md--GridColumnGap: var(--pf-t--global--spacer--md);
```
Comment (mcoker) [inline src/patternfly/components/Form/form.scss:35]:
```suggestion
  --#{$form}--m-horizontal__group-label--md--GridColumnGap: var(-pf-t--global--spacer--gap--group--horizontal);
```

File: src/patternfly/components/Form/form.scss [hunk 12]
```diff
@@ -85,15 +85,14 @@ $pf-v6-c-form--m-horizontal--breakpoint-map: build-breakpoint-map("sm", "md", "l
   --#{$form}__group-label-help--FontSize: var(--pf-t--global--font--size--body--sm);
 
   // Form group label info
-  --#{$form}__group-label--m-info--Gap: var(--pf-t--global--spacer--sm);
+  --#{$form}__group-label--m-info--Gap: var(--pf-t--global--spacer--gap--text-to-element--default);
```
Comment (mcoker) [inline src/patternfly/components/Form/form.scss:88]:
```suggestion
  --#{$form}__group-label--m-info--Gap: var(--pf-t--global--spacer--gap--group--horizontal);
```

File: src/patternfly/components/Form/form.scss [hunk 13]
```diff
@@ -85,15 +85,14 @@ $pf-v6-c-form--m-horizontal--breakpoint-map: build-breakpoint-map("sm", "md", "l
...
   --#{$form}__group-label-info--FontSize: var(--pf-t--global--font--size--body--sm);
-  --#{$form}--m-horizontal__group-label--m-info--Gap: var(--pf-t--global--spacer--sm);
+  --#{$form}--m-horizontal__group-label--m-info--Gap: var(--pf-t--global--spacer--gap--text-to-element--default);
```
Comment (mcoker) [inline src/patternfly/components/Form/form.scss:90]:
```suggestion
  --#{$form}--m-horizontal__group-label--m-info--Gap: var(--pf-t--global--spacer--gap--group--vertical);
```

File: src/patternfly/components/FormControl/form-control.scss [hunk 14]
```diff
@@ -100,7 +100,7 @@
   --#{$form-control}--m-error__icon--m-status--Color: var(--pf-t--global--icon--color--status--danger--default);
 
   // Form control utilities
-  --#{$form-control}__utilities--Gap: var(--pf-t--global--spacer--sm);
+  --#{$form-control}__utilities--Gap: var(--pf-t--global--spacer--gap--text-to-element--default);
```
Comment (mcoker) [inline src/patternfly/components/FormControl/form-control.scss:103]:
```suggestion
  --#{$form-control}__utilities--Gap: var(--pf-t--global--spacer--gap--group--horizontal);
```

File: src/patternfly/components/Hint/hint.scss [hunk 15]
```diff
@@ -1,7 +1,7 @@
 @use '../../sass-utilities' as *;
 
 @include pf-root($hint) {
-  --#{$hint}--GridRowGap: var(--pf-t--global--spacer--sm);
+  --#{$hint}--GridRowGap: var(--pf-t--global--spacer--gap--group--vertical);
```
Comment (mcoker) [inline src/patternfly/components/Hint/hint.scss:4]: Per @lboehling this is a good candidate for a more content specific token, so we can use a generic t-shirt spacer for now.

```suggestion
  --#{$hint}--GridRowGap: var(--pf-t--global--spacer--sm);
```

**Review by mcoker — COMMENTED** [review]
@thatblindgeye I met with @lboehling and did a group review and had these suggestions. Let me know if you'd like to go ahead and make the changes (and as long as everything works, we can just merge) or if you'd prefer merge this as-is and we can follow up post-v6 - either way is cool with me!

File: src/patternfly/components/ClipboardCopy/clipboard-copy.scss (hunk 7 above)
Comment (thatblindgeye) [reply src/patternfly/components/ClipboardCopy/clipboard-copy.scss:31]: Just to confirm, did you mean update teh MarginInlineStart token to the text-to-element spacer, or update the actions--Gap?

File: src/patternfly/components/CodeEditor/code-editor.scss [hunk 16]
```diff
@@ -3,7 +3,7 @@
 @include pf-root($code-editor) {
   // controls
   --#{$code-editor}__controls--PaddingInlineStart: var(--pf-t--global--spacer--sm);
-  --#{$code-editor}__controls--Gap: var(--pf-t--global--spacer--gap--control-to-control--plain);
+  --#{$code-editor}__controls--Gap: var(--pf-t--global--spacer--gap--action-to-action--plain);
```
Comment (mcoker) [inline src/patternfly/components/CodeEditor/code-editor.scss:6]: Updated to match the suggestion in https://github.com/patternfly/patternfly/pull/6957#discussion_r1747657764. Looks like a typo, `--pf-t--global--spacer--gap--control-to-control--plain` isn't a valid token.

File: src/patternfly/components/Form/form.scss [hunk 17]
```diff
@@ -29,10 +29,10 @@ $pf-v6-c-form--m-horizontal--breakpoint-map: build-breakpoint-map("sm", "md", "l
   --#{$form}--GridGap: var(--pf-t--global--spacer--gap--group-to-group--vertical);
 
   // Group
-  --#{$form}__group--Gap: var(--pf-t--global--spacer--gap--group--horizontal);
+  --#{$form}__group--Gap: var(--pf-t--global--spacer--gap--group--vertical);
```
Comment (mcoker) [inline src/patternfly/components/Form/form.scss:32]: Not sure why this was updated to `--horizontal`. The default layout for `__group` is a stack/vertical layout. Just reverting to what was there when we left some feedback around this block - https://github.com/patternfly/patternfly/pull/6957#discussion_r1747670189

File: src/patternfly/components/Form/form.scss [hunk 18]
```diff
@@ -87,7 +87,7 @@ $pf-v6-c-form--m-horizontal--breakpoint-map: build-breakpoint-map("sm", "md", "l
   // Form group label info
   --#{$form}__group-label--m-info--Gap: var(--pf-t--global--spacer--gap--horizontal);
   --#{$form}__group-label-info--FontSize: var(--pf-t--global--font--size--body--sm);
-  --#{$form}--m-horizontal__group-label--m-info--Gap: var(--pf-t--global--spacer--gap--vertical);
+  --#{$form}--m-horizontal__group-label--m-info--Gap: var(--pf-t--global--spacer--gap--group--vertical);
```
Comment (mcoker) [inline src/patternfly/components/Form/form.scss:90]: Applying suggestion from https://github.com/patternfly/patternfly/pull/6957#discussion_r1747672873. Looks like a typo, `group--` was left off (`--pf-t--global--spacer--gap--vertical` isn't a valid token)

File: src/patternfly/components/FormControl/form-control.scss [hunk 19]
```diff
@@ -100,7 +100,7 @@
   --#{$form-control}--m-error__icon--m-status--Color: var(--pf-t--global--icon--color--status--danger--default);
 
   // Form control utilities
-  --#{$form-control}__utilities--Gap: var(--pf-t--global--spacer--gap--horizontal);
+  --#{$form-control}__utilities--Gap: var(--pf-t--global--spacer--gap--group--horizontal);
```
Comment (mcoker) [inline src/patternfly/components/FormControl/form-control.scss:103]: Applying suggestion from https://github.com/patternfly/patternfly/pull/6957#discussion_r1747678586. Looks like a typo, `--pf-t--global--spacer--gap--horizontal` isn't a valid token.

**Review by mcoker — APPROVED** [review]

**Pull request merged**